app/controllers/shelters_controller.rb

```diff
@@ -1,19 +1,36 @@
 class SheltersController < ApplicationController
+  include ActionView::Helpers::TextHelper
+
   def index
     @shelters = Shelter.all
   end
 
   def show
     @shelter = Shelter.find(params[:id])
     @reviews = @shelter.reviews
+    @pet_counter = @shelter.pets.count
+    @average_rating = @reviews.average(:rating)
+    @shelter_applications = Array.new
+    @shelter.pets.each do |pet|
```

Comment:
Awesome job! I think we can use a .reduce or a .map here to create our new array so we can call `.flatten!` on the end  of the do block like you were trying during pairing

Reply:
That is so true, something to explore during refactoring.